### ISSUE TITLE
if: tmds: Remove clock from interface

### ISSUE THREAD

**Issue by sbobrowicz** [Digilent/vivado-library#2]
The clock should be removed from this interface. It will then be up to IP cores to declare the TMDS clock as a differential clock interface with the ASSOCIATED_BUSIF properly set. This should help clock constraints propagate through the tools more smoothly.

**Comment by elodg** [comment]
What is the exact issue you are seeing? I never had problems in Vivado with clock propagation. If there is no clock defined outside, the DVI maximum 165MHz is constrained by the core. One could override this by constraining the clock on the top-level port. In both cases it should propagate through.
I am not entirely clear on what ASSOCIATED_BUSIF does exactly. It helps with automation, but not sure what good would it do in our case.

**Comment by elodg** [comment]
Double-checked with the latest 1.6 version of the core. Constraining in top-level works, added example to documentation. I see no reason to remove the constraints from the core, nor to take it out from the TMDS interface.

**Comment by sbobrowicz** [comment]
I haven't experienced any clock propagation issues with the interface recently either, so maybe the initial post here was a bit misleading... Let me try again. My main motivation for raising this issue is to fall in line with how Xilinx tends to organize their interfaces in their IP cores. If a signal is going to be treated as a clock (pushed onto the BUFG network, input or output of a CMT, etc.), they tend to define the clock as a separate "clock" interface and then set the ASSOCIATED_BUSIF parameter for the clock interface to the name of the interface it is associated with (for example, "TMDS_0"). This tells IPI that the interface is on that clock domain. Sure, downstream tools will figure this out during the build process, but this provides some extra information that can propagate around at the IPI level. I think following Xilinx's lead on this would be a good design practice moving forward.

...But that said, I don't think there is much to gain from changing this interface now, and its not likely to cause any problems in the future either. I'll close this issue.

**Comment by sbobrowicz** [comment]
FYI, I just had the clock propagation issue crop up on me again, and figured out what was causing it. Our master XDC clock constraints all include the -add option, which indicates that if a clock already exists on the specified port, it should NOT be deleted. This would be useful if one wanted to analyze two different frequencies on a particular clock, but that doesn't seem like the typical use case, and certainly isn't the case here. I ran into this problem because I based the clock creation command I used for the TMDS clock on the command used in our ZYBO master XDC for the onboard 125MHz clock. 

I plan on removing the -add option from our create_clock commands in our master XDC since it doesn't seem like an appropriate default option.

Also note that overriding the clock seems to cause a critical warning. I will create a separate issue to resolve this critical warning.

**Comment by sbobrowicz** [comment]
Nevermind on the critical warning... it turns out removing the -name option as well silences that too. No additional issue will be opened

**Comment by elodg** [comment]
I commented out create_clock in dvi2rgb a while ago because of another issue:
# Constraining it here, even if overridden in the top-level project
# results in [DRC 23-20] Rule violation (PDRC-34) for the maximum MMCM VCO frequency.

In any case, if constrained in the top xdc, it should override the IP xdc.
So it makes sense to be without -add.
